### PR TITLE
Fix boto3 version

### PR DESCRIPTION
**Pull request by rcgottlieb**
## Changes proposed in this pull request:

- Updated Dockerfile and requirements files to use the current boto3 version
-
-

## Things to check

- For any logging statements, is there any chance that they could be logging sensitive data?
- Are log statements using a logging library with a logging level set? Setting a logging level means that log statements "below" that level will not be written to the output. For example, if the logging level is set to `INFO` and debugging statements are written with `log.debug` or similar, then they won't be written to the otput, which can prevent unintentional leaks of sensitive data.

## Security considerations

None

### CODE REVIEW

File: pip-tools/dev-requirements.in [hunk 1]
```diff
@@ -1,6 +1,6 @@
 -r ../requirements.txt
 black
-boto3-stubs[iam,cloudfront,elbv2]
+boto3-stubs[iam,cloudfront,elbv2]>1.34.33
```
Comment (markdboyd) [inline pip-tools/dev-requirements.in:3]: I'm curious: did we need to specify a version constraint here? Or would `upgrade-requirements` have upgraded it either way?

Comment (rcgottlieb) [reply pip-tools/dev-requirements.in:3]: thanks, I meant to remove that, I thought it was needed but meant to test without it and forgot. I don't think it's needed.

File: pip-tools/requirements.in [hunk 2]
```diff
@@ -1,6 +1,6 @@
 acme
 alembic
-boto3
+boto3>1.34.33
```
Comment (markdboyd) [inline pip-tools/requirements.in:3]: Same comment here as above

File: runtime.txt [hunk 3]
```diff
@@ -1 +1 @@
-3.8.x
+3.11.x
```
Comment (markdboyd) [inline runtime.txt:1]: Should this be 3.9.x if the docker tools target 3.9.x?

File: pip-tools/requirements.in (hunk 2 above)
Comment (rcgottlieb) [reply pip-tools/requirements.in]: yes, and done

**Review by markdboyd — APPROVED** [review]

**Pull request merged**